install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -1699,13 +1704,204 @@ dashboards:
 
 # -- Configure Istio proxy options.
 proxy:
+
   prometheus:
+    # -- Deprecated in favor of envoy.prometheus.enabled
     enabled: true
-    port: "9964"
+    # -- Deprecated in favor of envoy.prometheus.port
+    port: ~
   # -- Regular expression matching compatible Istio sidecar istio-proxy
   # container image names
   sidecarImageRegex: "cilium/istio_proxy"
 
+# -- Configure Cilium Envoy options.
+envoy:
+  # Enable Envoy Proxy in standalone DaemonSet.
+  enabled: false
+
+  # Location of unix sockets for communication between Agent & Envoy.
+  socketDir:
+    # Socket directory within the envoy & agent containers
+    containers: "/var/run/cilium/envoy/sockets"
+    # Socket directory on the host
+    host: "/var/run/cilium/envoy/sockets"
+
+  log:
+    # The format string to use for laying out the log message metadata of Envoy.
+    format: "[%Y-%m-%d %T.%e][%t][%l][%n] [%g:%#] %v"
+    # Path to a separate Envoy log file, if any. Defaults to /dev/stdout.
+    path: ""
+
+  # Time in seconds after which a TCP connection attempt times out
+  connectTimeoutSeconds: 2
+  # ProxyMaxRequestsPerConnection specifies the max_requests_per_connection setting for Envoy
+  maxRequestsPerConnection: 0
+  # Set Envoy HTTP option max_connection_duration seconds. Default 0 (disable)
+  maxConnectionDurationSeconds: 0
+
+  # -- Envoy container image.
+  image:
+    override: ~
+    repository: "quay.io/cilium/cilium-envoy"
+    tag: "v1.25.6-4350471813b173839df78f7a1ea5d77b5cdf714b"
+    pullPolicy: "${PULL_POLICY}"
+    digest: "sha256:5d03695af25448768062fa42bffec7dbaa970f0d2b320d39e60b0a12f45027e8"
+    useDigest: true
+
+  # -- Additional containers added to the cilium Envoy DaemonSet.
+  extraContainers: []
```

Comment:
Can we get rid of as many "extra$FOO" options as possible? Once shipped, they're impossible to remove.

Reply:
@squeed good point. but they are kind of already shipped: it's already possible to configure them for the `cilium` (agent) DaemonSet - where the Envoy L7 Proxy was part of.

Reply:
Mmm, I see what you're saying, but unless we know of a good use case for extraContaners et al. with Envoy, then I'd vote for "remove first, wait for someone to complain". If it somehow breaks their workflow they can stick with embedded until we add what's needed.

Reply:
@squeed "unfortunately" there are use cases to keep them.

* args & env: there are several command line options provided by Envoy. they might be used by users and or us (e.g. for analysis purposes (e.g. mentioned by @sayboras )
* volumes, mounts, ...: with the flexibility of exposing the Envoy config directly via `CiliumEnvoyConfig` users can leverage secret features (SDS) where the secrets are provided by the filesystem (also possible via environment variables) (e.g. mentioned by @jrajahalme )
* containers: might be the less used feature - but at some point its also worth to keep the configuration possibilities for the various cilium components (agent, operator, envoy, ...) aligned.